src/kontrol/solc_to_k.py

```diff
@@ -567,13 +623,39 @@ def method_sentences(self) -> list[KSentence]:
         res.extend(method.selector_alias_rule for method in self.methods)
         return res if len(res) > 1 else []
 
+    @staticmethod
+    def arg_name(input: Input) -> str:
+        return f'V{input.idx}_{input.name.replace("-", "_")}'
+
+    @staticmethod
+    def make_single_type(input: Input) -> KApply:
+        input_name = Contract.arg_name(input)
+        input_type = input.type
+        return KEVM.abi_type(input_type, KVariable(input_name))
+
+    @staticmethod
+    def recurse_components(components: list[Input]) -> KApply:
```

Comment:
```suggestion
    def recurse_components(components: Iterable[Input]) -> KApply:
```

Reply:
Thanks! I moved it to the `Input` class, together with the `_make_single_type(input: Input) -> KApply` function that I think now belongs there too: https://github.com/runtimeverification/kontrol/commit/0979c9406210e848e36a7ccf6ee141d3ed8b0ed8.